repo/fsrepo/fsrepo.go

```diff
@@ -387,6 +389,35 @@ func (r *FSRepo) SetAPIAddr(addr ma.Multiaddr) error {
 	return err
 }
 
+// SetGatewayAddr writes the Gateway Addr to the /gateway file.
+func (r *FSRepo) SetGatewayAddr(addr net.Addr) error {
+	// Create a temp file to write the address, so that we don't leave empty file when the
+	// program crashes after creating the file.
+	f, err := os.Create(filepath.Join(r.path, "."+gatewayFile+".tmp"))
+	if err != nil {
+		return err
+	}
+
+	if _, err = fmt.Fprintf(f, "http://%s", addr.String()); err != nil {
```

Comment:
Please don't use assignement if you use inline declarations, this is against go conventions and break programers (me) expectations.
Either don't use inline declarations:
```suggestion
	_, err = fmt.Fprintf(f, "http://%s", addr.String())
	if err != nil {
```
Or shadow the previous declaration:
```suggestion
	if _, err := fmt.Fprintf(f, "http://%s", addr.String()); err != nil {
```
Same for all other ones.